users/services.py

```
@@ -0,0 +1,20 @@
+import base64
+
+import boto3
+from django.conf import settings
+
+
+def upload_photo_to_s3(image_base64, user_id):
+    filename = f"{user_id}.jpg"
+    s3_client = boto3.client("s3")
+    try:
+        image_data = base64.b64decode(image_base64)
+        s3_client.put_object(
+            Bucket=settings.AWS_S3_PHOTO_BUCKET_NAME,
+            Key=filename,
+            Body=image_data,
+            ContentType="image/jpeg",
+        )
+    except Exception:
```

This will silence any error so we won't know they are occurring. If there are certain known exceptions we want to ignore, we should catch those explicitly. Otherwise, we should propagate the errors to sentry at a minimum and are potentially OK failing hard if they occur, especially since we will return a failing status code to the mobile in that case anyway.

---

I'd like to keep it to a caught exception (as opposed to failing hard) so that we can still return a helpful status code to mobile, however I do agree that this should bubble up to Sentry. 

I've made some tweaks to capture this exception to Sentry ([2883be5](https://github.com/dimagi/connect-id/pull/97/commits/2883be54ce7bb41c935a9e505479729706979dce)).

---

> we can still return a helpful status code to mobile
Can you say more about this? Specifically the ways this is more helpful. I would have imagined that a generic "this failed" with a 400 error would not have more info than  a 500 (both convey that the request failed), and a 500 seemed slightly more correct, since it represents an unexpected server error rather than the phone having made a bad request. If we could give the phone useful info to resolve the situation, I agree, but since we don't actually know what went wrong, a 500 feels as right (and 5xx implying server error rather than 4xx implying client error).

This solution addresses the monitoring issue, but doesn't seem better than the simpler version, unless we wanted the rest of the code in the view to execute.

---

> Can you say more about this?

What I mean by helpful is that by returning a predictable/static error code (as opposed to leaving the exception uncaught), mobile can then use this error code to present a translated helpful message to the user based on what code was received. This feels slightly more helpful than having the uncaught exception simply respond with the server error page template. Additionally, this also feels more consistent and in line with the other changes we've been making on ConnectID to always returned a predictable error code if something went wrong.

In saying the above, that was moreso focused around catching the exception rather than leaving it uncaught. Regarding your thought to have the response be a 500 instead of a 400, I am happy to make this change and can see your point on how this could be considered a server error rather than a client error.

---

> Regarding your thought to have the response be a 500 instead of a 400, I am happy to make this change and can see your point on how this could be considered a server error rather than a client error.

Happy to go forward with this. The error reporting was the big thing. We can always change this later if required.

---

Okay cool, I've updated the status code to be 500 now ([ad3aa01](https://github.com/dimagi/connect-id/pull/97/commits/ad3aa011535ba729eeecd02388961c6881093d4b)).